### PR TITLE
Add arg "return_raw"

### PR DESCRIPTION
**Pull request by donglihe-hub**
## What does this PR do?

1. Preprocessor.load_data can now return raw txt data.
2. [**API change**] Change read_libmultilabel_format and read_libsvm_format from public to private
3. Improve code readability by unifying the internal use of variables, i.e., data["text"] and data["label"] become data["x"] and data["y"] respectively.
4. Modify the gridsearch tutorial by changing the way it reads raw txt data.

## Test CLI & API (`bash tests/autotest.sh`)
Test APIs used by main.py.
- [ ] Test Pass
  - (Copy and paste the last outputted line here.)
- [x] Not Applicable (i.e., the PR does not include API changes.)

## Check API Document
If any new APIs are added, please check if the description of the APIs is added to API document. 
- [x] API document is updated ([linear](https://www.csie.ntu.edu.tw/~cjlin/libmultilabel/api/linear.html), [nn](https://www.csie.ntu.edu.tw/~cjlin/libmultilabel/api/nn.html))
- [ ] Not Applicable (i.e., the PR does not include API changes.)

## Test quickstart & API (`bash tests/docs/test_changed_document.sh`)
If any APIs in quickstarts or tutorials are modified, please run this test to check if the current examples can run correctly after the modified APIs are released.

**FAILED**  python3 docs/examples/plot_linear_gridsearch_tutorial.py
**FAILED**  python3 docs/examples/plot_dataset_tutorial.py
PASSED  python3 docs/examples/plot_linear_quickstart.py
PASSED  python3 docs/examples/plot_KimCNN_quickstart.py
PASSED  python3 docs/examples/plot_bert_quickstart.py

### CODE REVIEW

File: libmultilabel/linear/preprocessor.py [hunk 1]
```diff
@@ -51,6 +56,7 @@ def load_data(
             label_file (str, optional): Path to a file holding all labels. Defaults to None.
             include_test_labels (bool, optional): Whether to include labels in the test dataset. Defaults to False.
             remove_no_label_data (bool, optional): Whether to remove training instances that have no labels. Defaults to False.
+            return_raw: Whether to return raw libmultilabel format data.
 
         Returns:
             dict[str, dict[str, sparse.csr_matrix]]: The training and test data, with keys 'train' and 'test' respectively.
```
Comment (cjlin1) [inline libmultilabel/linear/preprocessor.py:62]: descriptions here should be modified as the returned object may not be a sparse csr matrix?

File: libmultilabel/linear/preprocessor.py [hunk 2]
```diff
@@ -51,6 +56,7 @@ def load_data(
             label_file (str, optional): Path to a file holding all labels. Defaults to None.
             include_test_labels (bool, optional): Whether to include labels in the test dataset. Defaults to False.
             remove_no_label_data (bool, optional): Whether to remove training instances that have no labels. Defaults to False.
+            return_raw: Whether to return raw libmultilabel format data.
```
Comment (cjlin1) [inline libmultilabel/linear/preprocessor.py:59]: can be more explict: True: ?? False: ??

File: libmultilabel/linear/preprocessor.py (hunk 1 above)
Comment (donglihe-hub) [reply libmultilabel/linear/preprocessor.py]: The returned type has been corrected

File: libmultilabel/linear/preprocessor.py (hunk 2 above)
Comment (donglihe-hub) [reply libmultilabel/linear/preprocessor.py]: Done

**Pull request closed without merging**